src/app/vendorEngramsXyzApi/vendorEngramsXyzService.ts

```diff
@@ -0,0 +1,122 @@
+import { VendorDrop, VendorEngramVendor, VendorDropType, ManifestVendor } from "./vendorDrops";
+import { loadingTracker } from "../ngimport-more";
+import { t } from 'i18next';
+
+export class VendorEngramsXyzService {
+  vendorMap: { [k: number]: VendorEngramVendor[] };
+  cachedResponse: VendorDrop[] | null;
+  lastUpdated: Date | null;
+
+  constructor() {
+    this.vendorMap = {};
+    this.vendorMap[ManifestVendor.AnaBray] = [VendorEngramVendor.AnaBray,
+      VendorEngramVendor.BraytechRWPMk_II,
+      VendorEngramVendor.IKELOS_HC_V1_0_1];
+    this.vendorMap[ManifestVendor.ArachJalaal] = [VendorEngramVendor.ArachJalaal];
+    this.vendorMap[ManifestVendor.AsherMir] = [VendorEngramVendor.AsherMir,
+      VendorEngramVendor.ManOWar];
+    this.vendorMap[ManifestVendor.Banshee44] = [VendorEngramVendor.Banshee44];
+    this.vendorMap[ManifestVendor.Benedict9940] = [VendorEngramVendor.Benedict9940];
+    this.vendorMap[ManifestVendor.CommanderZavala] = [VendorEngramVendor.CommanderZavala];
+    this.vendorMap[ManifestVendor.DevrimKay] = [VendorEngramVendor.DevrimKay,
+      VendorEngramVendor.MidaMiniTool];
+    this.vendorMap[ManifestVendor.TyraKarn] = [VendorEngramVendor.Drang];
+    this.vendorMap[ManifestVendor.ExecutorHideo] = [VendorEngramVendor.ExecutorHideo];
+    this.vendorMap[ManifestVendor.Failsafe] = [VendorEngramVendor.Failsafe];
+    this.vendorMap[ManifestVendor.IkoraRey] = [VendorEngramVendor.IkoraRey];
+    this.vendorMap[ManifestVendor.Lakshmi2] = [VendorEngramVendor.Lakshmi2];
+    this.vendorMap[ManifestVendor.LordSaladin] = [VendorEngramVendor.LordSaladin];
+    this.vendorMap[ManifestVendor.LordShaxx] = [VendorEngramVendor.LordShaxx];
+    this.vendorMap[ManifestVendor.Sloane] = [VendorEngramVendor.Sloane];
+    this.vendorMap[ManifestVendor.TheEmissary_TRIALS0] = [VendorEngramVendor.TheEmissary];
+    this.vendorMap[ManifestVendor.TheEmissary_TRIALS1] = [VendorEngramVendor.TheEmissary];
+    this.vendorMap[ManifestVendor.TheEmissary_TRIALS2] = [VendorEngramVendor.TheEmissary];
+    this.vendorMap[ManifestVendor.TheEmissary_TRIALS3] = [VendorEngramVendor.TheEmissary];
+    this.vendorMap[ManifestVendor.TheEmissary_TRIALS4] = [VendorEngramVendor.TheEmissary];
+  }
+
+  handleVendorEngramsErrors(response: Response) {
+    if (response.status !== 200) {
+      throw new Error(t('VendorEngramsXyz.ServiceCallError'));
+    }
+
+    return response.json();
+  }
+
+  lastUpdatedInPastFifteenMinutes(): boolean {
```

Comment:
If you're going to hardcode 15 minutes into the name of this method, you should also provide the interval for the code that polls it.